dataset.py

```diff
@@ -10,82 +13,47 @@
-
-        This creates a training set where each row is a time series of a
-        specific macroeconomic indicator for a specific country. The `X` table
-        includes the time series from 1972 to 2006, and the 'Y' table includes
-        the time series values for 2007. Missing values are coded as NaNs.
-
-        X and Y only include rows for which we need to make submissions for the
-        competition. Future iterations will include more rows to use as
-        features.
-
-        Returns:
-           X (pd.DataFrame): features for prediction
-           Y (pd.Series): targets for prediction
-        """
-
-        # Select rows for prediction only
-        X = self._train.loc[self._submit_rows.index]
-
-        # Select and rename columns
-        X = X.iloc[:, :-3]
-        X = X.rename(lambda x: int(x.split(' ')[0]), axis=1)
-
-        # Split prediction and target
-        Y = X.iloc[:, -1]  # 2007
-        X = X.iloc[:, :-1]  # 1972:2006
-
-        return X, Y
-
-
-    def preprocess_for_viz(self):
-        """Preprocess the data for visualization.
-
-        Selects rows for prediction and renames columns.
-        """
-
-        # Select rows for prediction only
-        X = self._train.loc[self._submit_rows.index]
-
-        # Select and rename columns
-        yrs = X.iloc[:, :-3]
-        names = X.iloc[:, -3:]
-        yrs = yrs.rename(lambda x: int(x.split(' ')[0]), axis=1)
-
-        df = pd.concat([yrs, names], axis=1)
-        gb = df.groupby('Series Name')
-
-        return gb
-
-
-    def evaluate(self, predictions):
-        """Check RMSE of predictions"""
-        _, Y = self.preprocess_simple()
-
-        return evaluation.RMSE(predictions, Y)
-
-
-if __name__=='__main__':
-
-    dataset = UNDevGoalsDataset()
-
-    X, Y = dataset.preprocess_simple()
-
-    print(X.describe())
-    print(Y.describe())
+    
+    def preprocess(self, pp_fn='preprocess_simple', pp_fn_kwargs={}):
+        """Preprocess data using function pp_fn (with additional kwargs if necessary) from preprocessing.py"""
+        """pp_fn is string with name of preprocessing model"""
+        
+        return eval('preprocessing.'+pp_fn+'(self._train, self._submit_rows.index, **pp_fn_kwargs)')
+    
+    
+    def predictions(self, model_name='status_quo_model', model_kwargs={}, pp_fn='preprocess_simple', pp_fn_kwargs={}):
+        """Return predictions from model_name when fit to data preprocessed by pp_fn"""
+        """pp_fn is string with name of preprocessing model, and model_name is name of prediction model"""
+        """Allows input of additional keyword parameters for model and preprocessing functions"""
+        
+        X,Y = self.preprocess(pp_fn, **pp_fn_kwargs)
+        return eval('models.'+model_name+'(X, **model_kwargs)') 
+    
+    
+    def prediction_error(self, error_fn='RMSE', model_name='status_quo_model', model_kwargs={}, pp_fn='preprocess_simple', pp_fn_kwargs={}):
```

Comment:
Same comment as above. I see that you're trying to unify the whole process of preprocessing+training+evaluation into just one function, but in the long run this will make things more complicated.

Preprocessing might take a long time, so we want the preprocessing function to return the preprocessed data, which we can then pass into a bunch of different models without having to preprocess again.

Training models might also take a long time, so we want the training function to return some pickled representation of the model so we can run the evaluation multiple times under different conditions (maybe to try different types of post-processing, or to evaluate performance on training vs. validation). We also need to save the predictions of the model to disk if we want to submit them to the competition.

Currently, this only allows you to measure RMSE by preprocessing, training, and predicting completely from scratch, even if you already have a preprocessed dataset and trained model.